Explorer/Assets/Scripts/ECS/Unity/Materials/Systems/StartMaterialsLoadingSystem.cs

```diff
@@ -163,16 +165,50 @@ private bool TryCreateGetTexturePromise(in TextureComponent? textureComponent, r
             // If component is being reused forget the previous promise
             ReleaseMaterial.TryAddAbortIntention(World, ref promise);
 
-            promise = Promise.Create(World, new GetTextureIntention
-            {
-                CommonArguments = new CommonLoadingArguments(textureComponentValue.Src, attempts: attemptsCount),
-                WrapMode = textureComponentValue.WrapMode,
-                FilterMode = textureComponentValue.FilterMode,
-            }, partitionComponent);
+            if (textureComponent.Value.IsVideoTexture)
+                promise = Promise.CreateFinalized(World, new GetTextureIntention
+                {
+                    CommonArguments = new CommonLoadingArguments(textureComponentValue.Src, attempts: attemptsCount),
+                    WrapMode = textureComponentValue.WrapMode,
+                    FilterMode = textureComponentValue.FilterMode,
+                    IsVideoTexture = textureComponentValue.IsVideoTexture,
+                    VideoPlayerEntity = textureComponentValue.VideoPlayerEntity,
+                }, partitionComponent, GetOrAddVideoTextureResult(textureComponentValue));
+            else
+                promise = Promise.Create(World, new GetTextureIntention
+                {
+                    CommonArguments = new CommonLoadingArguments(textureComponentValue.Src, attempts: attemptsCount),
+                    WrapMode = textureComponentValue.WrapMode,
+                    FilterMode = textureComponentValue.FilterMode,
+                    IsVideoTexture = textureComponentValue.IsVideoTexture,
+                    VideoPlayerEntity = textureComponentValue.VideoPlayerEntity,
+                }, partitionComponent);
 
             return true;
         }
 
+        private StreamableLoadingResult<Texture2D>? GetOrAddVideoTextureResult(TextureComponent textureComponent)
+        {
+            Texture2D videoTexture = null;
+
+            if (entitiesMap.TryGetValue(textureComponent.VideoPlayerEntity, out Entity videoPlayerEntity) && World.IsAlive(videoPlayerEntity))
+            {
+                if (World.Has<VideoTextureComponent>(videoPlayerEntity))
+                    videoTexture = World.Get<VideoTextureComponent>(videoPlayerEntity).Texture;
+                else
+                {
+                    var videoTextureComponent = new VideoTextureComponent(videoTexturesPool);
+                    World.Add(videoPlayerEntity, videoTextureComponent);
+
+                    videoTexture = videoTextureComponent.Texture;
+                }
+            }
+            else
+                ReportHub.LogError(ReportCategory.VIDEO_PLAYER, $"Entity {textureComponent.VideoPlayerEntity} not found!. VideoTexture will not be created.");
+
+            return new StreamableLoadingResult<Texture2D>(videoTexture);
```

Comment:
This a constructor for successful result: the branch with null should create a result with an exception instead